### PR TITLE
NFS: add preliminary nfs4 support

### PR DESCRIPTION
**Pull request by lrusak**
This adds NFSv4 support to Kodi's vfs.

This works by using a setting `nfs_set_version` after a context has been created.

NFSv4 also operates on the root always, this make it so there is only one context per hostname/ip. This also makes it so that we don't have to call `mount_getexports`. We can simply test if the file exists using `nfs_stat64`.

I'm not a big fan of how the nfs vfs is written and this seems like a bit of a shoehorn to get it working. I also don't like the new setting. This doesn't really allow switching protocols without restarting kodi (maybe a setting callback could be added in the future?).

If I had the time I would rather like to implement NFSv4 into a separate protocol or to add the ability to set the protocol version in `CGUIDialogNetworkSetup`. This PR is probably good enough for now though and would be a nice addition for Nexus.

I'm not sure if @sahlberg has any other pointer or maybe he can correct me if I got some facts wrong (I really have no idea about NFS).

### CODE REVIEW

**Comment by lrusak** [comment]
I forgot to say that the new setting is in the same location as the SMB version setting. I figured this was the best place for now.

I'm not sure who would be able to test this PR?

**Comment by fuzzard** [comment]
Tested locally, seems to be doing as mentioned on the tin

```
2022-07-12 15:16:21.687 T:7314195   DEBUG <general>: NFS: Context for 192.168.1.101/ not open - get a new context.
2022-07-12 15:16:21.687 T:7314195   DEBUG <general>: NFS: version: 4
2022-07-12 15:16:21.712 T:7314195   DEBUG <general>: NFS: Connected to server 192.168.1.101 and export /
2022-07-12 15:16:21.712 T:7314195   DEBUG <general>: NFS: chunks: r/w 0/0
2022-07-12 15:16:21.716 T:7314195   DEBUG <general>: NFS: Context for 192.168.1.102/ not open - get a new context.
2022-07-12 15:16:21.716 T:7314195   DEBUG <general>: NFS: version: 4
2022-07-12 15:16:21.734 T:7314195   DEBUG <general>: NFS: Connected to server 192.168.1.102 and export /
2022-07-12 15:16:21.734 T:7314195   DEBUG <general>: NFS: chunks: r/w 0/0
```

Was able to add an nfs4 mount, and play back media.

Tested disabling nfs4 on one of the NAS devices (192.168.1.101), so it only would run on nfs3. The mount was an existing one created long ago. It failed to open the mount (as expected i guess)

```
2022-07-12 15:26:05.244 T:7322953   DEBUG <general>: NFS: Context for 192.168.1.101/ not open - get a new context.
2022-07-12 15:26:05.244 T:7322953   DEBUG <general>: NFS: version: 4
2022-07-12 15:26:05.262 T:7322953   ERROR <general>: NFS: Failed to mount nfs share: / (mount_cb: )
2022-07-12 15:26:05.262 T:7322953   ERROR <general>: GetDirectory - Error getting nfs://192.168.1.101/volume2/TV/
2022-07-12 15:26:05.262 T:7322953   DEBUG <general>: Thread waiting 0x3334f4000 terminating
2022-07-12 15:26:05.265 T:7321728   ERROR <general>: CGUIMediaWindow::GetDirectory(nfs://192.168.1.101/volume2/TV/) failed
2022-07-12 15:26:05.266 T:7321728   DEBUG <general>: CGUIMediaWindow::GetDirectory (sources://video/)
```

I know you are going to hate this, but i think it would be better to either fallback to an older nfs version if it fails against nfs4 like above, or add a new protocol to handle nfs4 explicitly (dont want to clobber existing nfs shares for users).

I know its probably a bit of a contrived setup (2 x NFS servers, 1 max version4.1, 1 max version 3), so do whatever you think i guess.

Short version, nfs 4 works, with the restrictions of either/or version support, not both.

File: xbmc/filesystem/NFSFile.cpp [hunk 1]
```diff
@@ -49,6 +50,13 @@
 
 using namespace XFILE;
 
+namespace
+{
+
+const auto SETTING_NFS_VERSION = "nfs.version";
```
Comment (howie-f) [inline xbmc/filesystem/NFSFile.cpp:56]: was it meant `constexpr auto` ?

**Comment by lrusak** [comment]
> I know you are going to hate this, but i think it would be better to either fallback to an older nfs version if it fails against nfs4 like above, or add a new protocol to handle nfs4 explicitly (dont want to clobber existing nfs shares for users).
> 

Like I said I know this change isn't ideal but it's really all the time I have. Doing anything more will likely require a bit of a rewrite of the nfs handling and the network connections setup stuff. This change adds functionality at no cost really.

File: xbmc/filesystem/NFSFile.cpp (hunk 1 above)
Comment (lrusak) [reply xbmc/filesystem/NFSFile.cpp:56]: sure

**Comment by enen92** [comment]
I guess this needs an update to the wiki due to the new setting. Pinging @KarellenX for reference
